infrastructure/fw-trace/fw-trace.py

```diff
@@ -0,0 +1,407 @@
+				allow_matched = False
+
+				for prot in item['allowed']:
+					if prot['IPProtocol'] == "all":
+						if str(verbose) <> "None" : print ('  Egress firewall rule "%s" allows all protocols and ports' % (item['name']))
+						allow_matched  = True
+						break
+				
+					matched_port = True
+					if ((prot['IPProtocol'] == "tcp") or (prot['IPProtocol'] == "udp")) and (prot['IPProtocol']  == protocol):
+						#checking to see if port matches any item is port list
+						matched_port = False
+						for p in prot['ports']:	
+							if p == port: 
+								matched_port = True
+								break
+							if '-' in p:				
+								if (int(port) >= int(p.partition('-')[0])) and (int(port) <= int(p.partition('-')[2])):
+									matched_port = True
+									break							
+														
+					if (protocol == prot['IPProtocol']) and matched_port :
+						if str(verbose) <> "None" : print ('  Egress firewall rule "%s" with allowed protocols %s matches the protocol %s:%s  ' % (item['name'], item['allowed'],protocol,port))
+						allow_matched  = True
+						break   
+
+			
+				if not allow_matched :
+					if str(verbose) <> "None" : print ('  Egress firewall rule "%s" does not allow or deny traffic on protocol %s:%s. Skipping to next firewall rule' % (item['name'], protocol, port))
+					continue
+				else:
+					print ('--->Egress firewall rule "%s" matched.  Traffic is allowed to egress to %s' % (item['name'],dst))
+					implicit_allow = False
+					break
+		
+			#check to see if protocol/port match denied list
+			if 'denied' in item:
+				denied_matched = False
+
+				for prot in item['denied']:
+					if prot['IPProtocol'] == "all":
+						if str(verbose) <> "None" : print ('  Egress firewall rule "%s" allows all protocols and ports' % (item['name']))
+						denied_matched  = True
+						break
+
+					matched_port = True
+					if ((prot['IPProtocol'] == "tcp") or (prot['IPProtocol'] == "udp")) and (prot['IPProtocol']  == protocol):
+						#checking to see if port matches any item is port list
+						matched_port = False
+						for p in prot['ports']:	
+							if p == port: 
+								matched_port = True
+								break
+							if '-' in p:				
+								if (int(port) >= int(p.partition('-')[0])) and (int(port) <= int(p.partition('-')[2])):
+									matched_port = True
+									break							
+														
+					if (protocol == prot['IPProtocol']) and matched_port :
+						if str(verbose) <> "None" : print ('  Egress firewall rule "%s" with allowed protocols %s matches the protocol %s:%s  ' % (item['name'], item['allowed'],protocol,port))
+						denied_matched  = True
+						break   
+
+			
+				if not denied_matched :
+					if str(verbose) <> "None" : print ('  Egress firewall rule "%s" does not allow or deny traffic on protocol %s:%s. Skipping to next firewall rule' % (item['name'], protocol, port))
+					continue
+				else:
+					print ('--->Egress firewall rule "%s" matched.  Traffic egress to %s is blocked' % (item['name'],dst))	
```

Comment:
It might be useful to have an overall "traffic is allowed" output? 
Something like :

Traffic Denied
--->Egress firewall rule "disallow-egress-tag-2" matched.  Traffic egress to 10.150.0.2 is blocked
--->Ingress firewall rule "allow-8000-ingress" matched.  Traffic is allowed to ingress to 10.150.0.2

or 

Traffic Allowed
--->Reached end of egress firewall rules. Traffic egress to 10.150.0.2 is implicitly allowed
--->Ingress firewall rule "allow-8000-ingress" matched.  Traffic is allowed to ingress to 10.150.0.2


Thanks